src/wayland-egldisplay.c

```diff
@@ -670,48 +674,103 @@ EGLBoolean wlEglTerminateHook(EGLDisplay dpy)
     return res;
 }
 
-static void getServerProtocolsInfo(struct wl_display *nativeDpy,
+static char* getNodeFromDrmDevice(dev_t dev) {
+    char *result = NULL;
+    drmDevice *drm_device;
+    if (drmGetDeviceFromDevId(dev, 0, &drm_device) == 0) {
+        if (drm_device->available_nodes & (1 << DRM_NODE_RENDER)) {
+            result = strdup(drm_device->nodes[DRM_NODE_RENDER]);
+        } else if (drm_device->available_nodes & (1 << DRM_NODE_PRIMARY)) {
+            result = strdup(drm_device->nodes[DRM_NODE_PRIMARY]);
+        }
+
+        drmFreeDevice(&drm_device);
+    }
+    return result;
+}
+
+static bool getServerProtocolsInfo(WlEglDisplay *dpy,
                                    WlServerProtocols *protocols)
 {
+    struct wl_display     *nativeDpy    = dpy->nativeDpy;
     struct wl_display     *wrapper      = NULL;
     struct wl_registry    *wlRegistry   = NULL;
     struct wl_event_queue *queue        = wl_display_create_queue(nativeDpy);
-    int                    ret          = 0;
+    int                    ret          = -1;
     const struct wl_registry_listener registryListener = {
         registry_handle_global_check_protocols,
         registry_handle_global_remove
     };
 
     if (queue == NULL) {
-        return;
+        goto out;
     }
 
     wrapper = wl_proxy_create_wrapper(nativeDpy);
+    if (!wrapper) {
+        goto out;
+    }
     wl_proxy_set_queue((struct wl_proxy *)wrapper, queue);
 
     /* Listen to wl_registry events and make a roundtrip in order to find the
      * wl_eglstream_display global object.
      */
     wlRegistry = wl_display_get_registry(wrapper);
+    if (!wlRegistry) {
+        goto out;
+    }
     wl_proxy_wrapper_destroy(wrapper); /* Done with wrapper */
```

Comment:
I think we can remove this (and the line below) since we're now destroying the wrapper at the end.